tslearn/neighbors.py

```diff
@@ -104,6 +159,8 @@ def cdist_fun(X, Xp):
 
         n_ts = X.shape[0]
         sample_range = numpy.arange(n_ts)[:, None]
+        ind = ind[
```

Comment:
What does this do?

Reply:
It will do the same as the previous code, but this is similar to how sklearn does it. This will result in differences when breaking ties (two or more samples having minimal distances), and therefore ensures that the output using an sklearn pipeline (without any tslearn classifiers) is the same as when we are using tslearn classifiers.

Tbh, I think this code is much less clear then the previous one, so it could definitely be reverted back.

Reply:
OK, then could you add that info in a comment?

Reply:
Agreed, I will document this better :) (bcs even I have troubles understanding it clearly now already)

Reply:
Ok so I've been more closely inspecting the code, and it does not only mimic the output of sklearn this way (TimeSeriesKNeighbor~(metric='euclidean') has the same results as feeding a distance matrix to sklearn.KNeighbors~(metric='euclidean')), it is also an optimization as `argpartition` happens in O(n) and then the sort happens on the k nearest neighbors in O(k log(k)), and k << n.